services/router/service.go

```diff
@@ -1,35 +1,67 @@
 package router
 
 import (
-	"Bete/routes/restaurant"
+	orderRouter "Bete/routes/order"
+	restaurantRouter "Bete/routes/restaurant"
+	userRouter "Bete/routes/user"
+	userService "Bete/services/user"
+	"context"
 	"fmt"
-	"github.com/go-chi/chi/v5"
-	"go.uber.org/fx"
 	"net/http"
 	"os"
+
+	"github.com/go-chi/chi/v5"
+	"go.uber.org/fx"
 )
 
 type routerService struct {
 	restaurantRouter restaurantRouter.Router
+	orderRouter      orderRouter.Router
+	userRouter       userRouter.Router
+	userService      userService.Service
 }
 
 type params struct {
 	fx.In
 
 	RestaurantRouter restaurantRouter.Router
+	OrderRouter      orderRouter.Router
+	UserRouter       userRouter.Router
+	UserService      userService.Service
 }
 
 func New(p params) Service {
 	return &routerService{
 		restaurantRouter: p.RestaurantRouter,
+		orderRouter:      p.OrderRouter,
+		userRouter:       p.UserRouter,
+		userService:      p.UserService,
 	}
 }
 
+func (s routerService) EnsureUser(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		type ContextKey string
+		const ContextTokenKey ContextKey = "Id"
```

Comment:
what is this type? remove and just add const anywhere _idKey

Reply:
This would throw a lint error: `should not use built-in type string as key for value; define your own type to avoid collisions (SA1029)`
ref 
https://go.dev/blog/context#TOC_3.2.:~:text=To%20avoid%20key%20collisions%2C%20userip%20defines%20an%20unexported%20type%20key%20and%20uses%20a%20value%20of%20this%20type%20as%20the%20context%20key

https://stackoverflow.com/questions/40891345/fix-should-not-use-basic-type-string-as-key-in-context-withvalue-golint